core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -731,14 +731,21 @@ abstract class AbstractControllerBrokerRequestBatch(config: KafkaConfig,
       else if (config.interBrokerProtocolVersion >= KAFKA_2_2_IV0) 1
       else 0
 
+    /**
+     * For StopReplica request version below 4, we rely on the isPartitionDeleted function to check a partition's delete flag,
```

Comment:
nit: For describing versions, I prefer using `<` or `<=` to make it crisp and obvious.

Reply:
Fixed.